src/main/java/com/github/cameltooling/lsp/internal/parser/CamelYamlDSLParser.java

```diff
@@ -177,20 +191,40 @@ private boolean isStartOfUri(int separatorPos, int pos, String line) {
 	}
 
 	public String getCorrespondingType(TextDocumentItem textDocumentItem, int lineNumber) {
-		for (int lineNo = lineNumber; lineNo >=0; lineNo--) {
-			String tempLine = parserFileHelperUtil.getLine(textDocumentItem, lineNo);
-			Map<?, ?> data = parseYaml(tempLine);
-			if (data != null) {
-				if (data.containsKey(TO_KEY)) {
-					return "to";
-				} else if (data.containsKey(FROM_KEY)) {
-					return "from";
-				} else if (data.containsKey(REST_KEY)) {
-					return null;
+		StringBuilder sb = new StringBuilder();
+		extractLines(textDocumentItem, new Position(lineNumber, 0), "-", sb, 0, false);
+
+		Yaml yaml = new Yaml(new SafeConstructor(new LoaderOptions()));
+		return getCorrespondingTypeFromUnknownElement(yaml.load(sb.toString()));
+	}
+
+	private static String getCorrespondingTypeFromUnknownElement(Object yamlElement) {
```

Comment:
We need to iterate here because now, as we are multi-line, we are not sure on which level we have to search for the camel uri.

It can be a simple `to: "aws:something"` or it can be a `uri` inside a `to` element.